scripts/commands/updateApiDocs.ts

```diff
@@ -207,9 +226,27 @@ zxMain(async () => {
     args.version,
     legacyReleaseNoteEntries,
     versionWithoutPatch,
+    args.historical,
   );
 });
 
+/**
+ * Deletes all the files in the folder, but preserves subfolders. That's important
+ * to avoid accidentally deleting other historical versions of the API.
+ *
+ * We delete files when regenerating API docs to capture when APIs have been deleted.
+ */
+async function rmFilesInFolder(
+  dir: string,
+  description: string,
+  historical: boolean,
+): Promise<void> {
+  console.log(`Deleting existing markdown for ${description}`);
+  historical
+    ? await $`find ${dir}/* -not -path "*release-notes*" | xargs rm -rf {}`
+    : await $`find ${dir}/* -not -path "*release-notes*" -not -path '*/[0-9]*' | xargs rm -rf {}`;
```

Comment:
This is safe to run on folders without historical versions, right? We're going to be adding historical versions soon to runtime and provider - I think it's fine to always use this code. Simpler and more future proof.

Reply:
Oh, I see. We cannot run the same command in both versions because:

- Line 246 run in a non-historical version, will remove all the historical folders.
- Line 247 run in a historical version won't remove anything because "-not -path '*/[0-9]*'" will be true always.

I can investigate if there's another method to use only one command.

Reply:
Ah. My original thinking was only to remove the _files_ in the folder and don't remove any subfolders. That should handle all the edge cases, including not deleting release notes.

Reply:
That's a good idea! I replaced the two lines for `find ${dir}/* -maxdepth 0 -type f | xargs rm -f {}` where we only remove files in the same depth (not files in any subfolder).